odoo/models.py

```diff
@@ -3360,6 +3357,146 @@ def read(self, fields=None, load='_classic_read'):
 
         return self._read_format(fnames=fields, load=load)
 
+    def fetch_field_translations(self, fname):
+        """ fetch translations in cache """
+        # TODO CWG: check read access right and rules:
+        # TODO CWG: deal with related translated fields
+        # TODO CWG: related field
+        field = self._fields[fname]
+        if not field.translate:
+            raise UserError(_('You can only fetch translations for translated fields'))
+        records = self
+
+        while field.related:
+            records = records.mapped(field.related[:field.related.rfind('.')])
+            field = field.related_field
+
+        cache = self.env.cache
+        try:
+            for record in records:
+                cache.get(record, field)
+            return
+        except Exception:
+            pass
+
+        records.check_access_rights('read')
+        records.check_field_access_rights('read', [fname])
+        records.check_access_rule('read')
+
+        records.flush_recordset([field.name])
+
+        query = f"""SELECT {fname}, id FROM {self._table} WHERE id in %s"""
+        cr = self.env.cr
+        langs = set(l[0] for l in self.env['res.lang'].get_installed())
+        langs.add('en_US')
+
+        for sub_ids in cr.split_for_in_conditions(records.ids):
+            cr.execute(query, (sub_ids,))
+            for row in cr.fetchall():
+                record = records.browse(row[1])
```

Comment:
@HydrionBurst 
To speed up the load of jsonb columns by psycopg2 you could register your own json loader based on [`ujson`](https://pypi.org/project/ujson/) 
```python

import ujson
# this could be done on the connection
psycopg2.extras.register_json(conn, loads=ujson.loads, name='jsonb')
# or on the cursor...
psycopg2.extras.register_json(cr, loads=ujson.loads, name='jsonb')
```

This will improve a lot the perf when loading translations on a large recordset. By the way, it would be also nice to use `ujson` for `Serialized' fields.

It